docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,85 @@
+
+Mozilla generally favors a rapid release model to long cycle times.  Firefox desktop adopted rapid releases over a decade ago and there is current work to move both the mobile applications and our services towards rapid releases.  Although there are some costs to rapid releases, as a company we've decided that we're willing to accept them.
+
+Application services should consider moving towards a rapid release model.  The same reasoning that led other teams to adopt rapid releases also applies to us.  Furthermore, the mobile teams are moving towards rapid releases which may depend on us also adopting them.  Finally, we are in the process of [moving towards a single monorepo for android code](https://docs.google.com/document/d/1VMr5rtn-2Ilb4DSjNoD6D9bk-RX1GGOin8Hpn9sqMbA/edit), and it's somewhat likely that the monorepo will contain the iOS code as will.  A monorepo would require that application services code is taken by consumer applications as soon as it's merged.  Reducing the amount of time between our releases and when applications take those releases brings us closer to that monorepo world.
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+    * Keep everything the same for a while and wait for other code to move into the monorepo.
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process
+        * Update our branches to match how moz-central: we would have a `main`, `beta` and `release` branch, with nightly being built off of main, as well as `esr-*` branches for older ESR releases.
+        * Automate a nightly release of our `main` branch.
+          * Create a commit that updates our version number to something like `[firefox-version].0.nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * The version string is composed of major, minor, and patch components.  This matches firefox-android, but not moz-central, which skips the minor component.  The reason to have a minor component is to support weekly releases for mobile apps.  They may choose to include new features on those apps, which may require us to uplift new features to our release version and that would be a minor version bump.
```

Comment:
I'm pretty conflicted about this one, maybe we could just have 2 components like firefox does.  Like I mention below, we would already be breaking from semver.

Reply:
If we're already breaking from semver we might as well change the version number format imo. Sticking with the 3 component version number might be confusing.

Reply:
I agree that major.minor is all we'd need. By the time we're doing point releases, I can't imagine ever backporting such a significant breaking change that we'd be uncomfortable with more than a minor version bump. The joys of a 4 week release cycle!